Cargo.toml

```diff
@@ -38,12 +45,16 @@ ethers-signers  = { git = "https://github.com/scroll-tech/ethers-rs.git", branch
 #ethers-etherscan = { git = "https://github.com/scroll-tech/ethers-rs.git", branch = "v2.0.7" }
 #ethers = { git = "https://github.com/scroll-tech/ethers-rs.git", branch = "v2.0.7" }
 [patch."https://github.com/privacy-scaling-explorations/halo2.git"]
-halo2_proofs = { git = "https://github.com/scroll-tech/halo2.git", branch = "v1.1"  }
+halo2_proofs = { git = "https://github.com/snarkify/halo2", branch = "snarkify/dev.scroll.alpha.2" }
 [patch."https://github.com/privacy-scaling-explorations/poseidon.git"]
 poseidon = { git = "https://github.com/scroll-tech/poseidon.git", branch = "main" }
 [patch."https://github.com/privacy-scaling-explorations/bls12_381"]
 bls12_381 = { git = "https://github.com/scroll-tech/bls12_381", branch = "feat/impl_scalar_field" }
 
+[patch."https://github.com/scroll-tech/poseidon-circuit.git"]
+poseidon-base = { package = "poseidon-base", git = "https://github.com/snarkify/scroll-poseidon-circuit", branch = "main" }
+hash-circuit = { package = "poseidon-circuit",  git = "https://github.com/snarkify/scroll-poseidon-circuit", branch = "main"  }
```

Comment:
if remove above two patches and the code still run, then we don't need to add them